test/ad/distributions.jl

```diff
@@ -1,586 +1,621 @@
+        #DistSpec(Gamma, (), [0.5]),
+        #DistSpec(Gamma, (1.0,), [0.5]),
+        #DistSpec(Gamma, (1.0, 2.0), [0.5]),
+
+        DistSpec(GeneralizedExtremeValue, (1.0, 1.0, 1.0), 0.5),
+        #DistSpec(GeneralizedExtremeValue, (1.0, 1.0, 1.0), [0.5]),
+
+        DistSpec(GeneralizedPareto, (), 0.5),
+        DistSpec(GeneralizedPareto, (1.0, 2.0), 0.5),
+        DistSpec(GeneralizedPareto, (0.0, 2.0, 3.0), 0.5),
+        #DistSpec(GeneralizedPareto, (), [0.5]),
+        #DistSpec(GeneralizedPareto, (1.0, 2.0), [0.5]),
+        #DistSpec(GeneralizedPareto, (0.0, 2.0, 3.0), [0.5]),
+
+        DistSpec(Gumbel, (), 0.5),
+        DistSpec(Gumbel, (1.0,), 0.5),
+        DistSpec(Gumbel, (1.0, 2.0), 0.5),
+        #DistSpec(Gumbel, (), [0.5]),
+        #DistSpec(Gumbel, (1.0,), [0.5]),
+        #DistSpec(Gumbel, (1.0, 2.0), [0.5]),
+
+        DistSpec(InverseGamma, (), 0.5),
+        DistSpec(InverseGamma, (1.0,), 0.5),
+        DistSpec(InverseGamma, (1.0, 2.0), 0.5),
+        #DistSpec(InverseGamma, (), [0.5]),
+        #DistSpec(InverseGamma, (1.0,), [0.5]),
+        #DistSpec(InverseGamma, (1.0, 2.0), [0.5]),
+
+        DistSpec(InverseGaussian, (), 0.5),
+        DistSpec(InverseGaussian, (1.0,), 0.5),
+        DistSpec(InverseGaussian, (1.0, 2.0), 0.5),
+        #DistSpec(InverseGaussian, (), [0.5]),
+        #DistSpec(InverseGaussian, (1.0,), [0.5]),
+        #DistSpec(InverseGaussian, (1.0, 2.0), [0.5]),
+
+        DistSpec(Kolmogorov, (), 0.5),
+        #DistSpec(Kolmogorov, (), [0.5]),
+
+        DistSpec(Laplace, (), 0.5),
+        DistSpec(Laplace, (1.0,), 0.5),
+        DistSpec(Laplace, (1.0, 2.0), 0.5),
+        #DistSpec(Laplace, (), [0.5]),
+        #DistSpec(Laplace, (1.0,), [0.5]),
+        #DistSpec(Laplace, (1.0, 2.0), [0.5]),
+
+        DistSpec(Levy, (), 0.5),
+        DistSpec(Levy, (0.0,), 0.5),
+        DistSpec(Levy, (0.0, 2.0), 0.5),
+        #DistSpec(Levy, (), [0.5]),
+        #DistSpec(Levy, (0.0,), [0.5]),
+        #DistSpec(Levy, (0.0, 2.0), [0.5]),
+
+        DistSpec((a, b) -> LocationScale(a, b, Normal()), (1.0, 2.0), 0.5),
+        #DistSpec((a, b) -> LocationScale(a, b, Normal()), (1.0, 2.0), [0.5]),
+
+        DistSpec(Logistic, (), 0.5),
+        DistSpec(Logistic, (1.0,), 0.5),
+        DistSpec(Logistic, (1.0, 2.0), 0.5),
+        #DistSpec(Logistic, (), [0.5]),
+        #DistSpec(Logistic, (1.0,), [0.5]),
+        #DistSpec(Logistic, (1.0, 2.0), [0.5]),
+
+        DistSpec(LogitNormal, (), 0.5),
+        DistSpec(LogitNormal, (1.0,), 0.5),
+        DistSpec(LogitNormal, (1.0, 2.0), 0.5),
+        #DistSpec(LogitNormal, (), [0.5]),
+        #DistSpec(LogitNormal, (1.0,), [0.5]),
+        #DistSpec(LogitNormal, (1.0, 2.0), [0.5]),
+
+        DistSpec(LogNormal, (), 0.5),
+        DistSpec(LogNormal, (1.0,), 0.5),
+        DistSpec(LogNormal, (1.0, 2.0), 0.5),
+        #DistSpec(LogNormal, (), [0.5]),
+        #DistSpec(LogNormal, (1.0,), [0.5]),
+        #DistSpec(LogNormal, (1.0, 2.0), [0.5]),
```

Comment:
Is there a reason why we have all these comments? Unless I'm completely mistaken, these are always identical to the lines above, right?

Reply:
I don't know why they exist (I just copied them since I thought maybe there's some reason). They are not identical (the difference seems to be that they use vector-valued samples/locations), but since we test the gradient with respect to vector-valued inputs in the end even the scalar samples end up as part of a vector. IMO the comments could just be removed. Would that be fine with you?

Reply:
Ah, sorry; my slightly sleep-deprived mind could not comprehend `[` and `]`. But yeah, that sounds good to me :+1: